experimental/list_formatter/src/list_formatter.rs

```diff
@@ -41,12 +41,18 @@ impl ListFormatter {
     }
 
     /// Returns a `Writeable` composed of the input `Writeable`s and the language-dependent
-    /// formatting. The first layer of fields contains `ListFormatter::element()` for input elements,
-    /// and `ListFormatter::literal()` for list literals.
-    pub fn format<'a, 'b, W: Writeable>(&'a self, values: &'b [W]) -> List<'a, 'b, W> {
+    /// formatting. The first layer of fields contains `ListFormatter::element()` for input
+    /// elements, and `ListFormatter::literal()` for list literals.
+    pub fn format<'a, W: Writeable + 'a, I: IntoIterator<Item = W>>(
```

Comment:
Thought: My only concern is that `Vec` implements `IntoIterator`, returning a [std::vec::IntoIter](https://doc.rust-lang.org/std/vec/struct.IntoIter.html), which means that people could pass a `vec` by value into this function, and then the `Clone` is not cheap since it clones the vector.  The correct thing should be to get a slice from the vec before passing it into the format function.

Reply:
yeah that was kinda my concern. with Iterators it's harder to pass in something cloneable that's also expensive to clone

Reply:
changed back to `Iterator`